mycore-base/src/main/java/org/mycore/common/MCRSessionResolver.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * This file is part of ***  M y C o R e  ***
+ * See http://www.mycore.de/ for details.
+ *
+ * MyCoRe is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MyCoRe is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MyCoRe.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.mycore.common;
+
+import java.io.Serializable;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.servlet.http.HttpSessionBindingEvent;
+import javax.servlet.http.HttpSessionBindingListener;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.mycore.frontend.servlets.MCRServlet;
+
+/**
+ * This Class will be stored in the a {@link javax.servlet.http.HttpSession} and can be used to resolve the
+ * {@link MCRSession}. We can not store {@link MCRSession} directly in the {@link javax.servlet.http.HttpSession}
+ * because values need to be {@link java.io.Serializable}.
+ */
+public final class MCRSessionResolver implements Serializable, HttpSessionBindingListener {
+
+    private static final ConcurrentHashMap<String, MCRSessionResolver> INSTANCE_MAP = new ConcurrentHashMap<>();
```

Comment:
No instance map needed. Object is to light-weighted for this overhead. It solely consists of a single String that is shared with the MCRSession.